### PR TITLE
Shortest path

### LINKED ISSUE #232: Shortest Path (Visualization)
## User story
1. As a Sales Manager
2. I want/need to talk to a person outside my organization and need to know who the people in between are
3. So that I can sell more products

## Acceptance criteria
* In Filtermenu, add subsection "shortest Path"
  * Let user select two nodes (start/end) -> input search field with expandable dropdown (-> like searchfuction #30 )
* Applied to both Hierarchical, as well as Graph layout
* In respective visualization: highlight start node, end node, and edges/nodes in-between
* use colors from  https://material-ui.com/de/customization/color/ (the Spielwiese can help you check look and feel)
* Highlighted path shows shortest-path
* global search for start/end
*  Filter applies for nodes in-between
* short means number of edges
* Example visualization:
image

## Definition of done
* Documentation conform with tsdoc
* Reviewed with 4 eyes principle
* Did not break other code
* Understandable documentation for developers

### LINKED ISSUE #269: Rename EntityVisualisationAttributes
The correct spelling is `EntityVisualizationAttributes`

### LINKED ISSUE #270: Installation of GDS Plugin
Installation of GDS Plugin with local run scripts, in deployment and in ci tests.

### PR DESCRIPTION
**Pull request by CatoLeanTruetschel**
Implements and closes #232, Closes #270, Closes #269

### CODE REVIEW

File: frontend/src/services/shortest-path/ShortestPathService.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* istanbul ignore file */
+
+import { injectable } from 'inversify';
+import 'reflect-metadata';
+import { QueryResult, ShortestPathQuery } from '../../shared/queries';
+import { CancellationToken } from '../../utils/CancellationToken';
+
+@injectable()
+export default abstract class ShortestPathService {
```
Comment (PianoRollRepresentation) [inline frontend/src/services/shortest-path/ShortestPathService.ts:9]: missing documentation

File: frontend/src/stores/colors/EntityStyle.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+export interface EntityStrokeStyle {
```
Comment (PianoRollRepresentation) [inline frontend/src/stores/colors/EntityStyle.ts:1]: missing documentation

File: frontend/src/stores/shortest-path/ShortestPathStateStore.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import SimpleStore from '../SimpleStore';
+import { ShortestPathState } from './ShortestPathState';
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (PianoRollRepresentation) [inline frontend/src/stores/shortest-path/ShortestPathStateStore.ts:4]: why is this necessary?

Comment (PianoRollRepresentation) [reply frontend/src/stores/shortest-path/ShortestPathStateStore.ts:4]: also add a short documentation what the store is responsible for

File: frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx [hunk 4]
```diff
@@ -6,14 +6,17 @@ import useService from '../../dependency-injection/useService';
 import QueryResultStore from '../../stores/QueryResultStore';
 import useObservable from '../../utils/useObservable';
 
+type nodeChangedCallback = (id: number | null) => void;
```
Comment (PianoRollRepresentation) [inline frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx:9]: use capital letter for the first letter since this is a custom type

**Review by PianoRollRepresentation — CHANGES_REQUESTED** [review]

**Review by joluj — APPROVED** [review]
LGTM.
Next time please rename files like `EntityStyleProviderImpl.ts` instead of deleting and creating it with a new name. Now we have lost the old versions and the information on who did what.

**Comment by CatoLeanTruetschel** [comment]
Known issue:
* If start/end node are subsidiary in the filter query-result, as they are not part of the filter, they must not have this flag set.

**Comment by CatoLeanTruetschel** [comment]
>Next time please rename files like EntityStyleProviderImpl.ts instead of deleting and creating it with a new name. Now we have lost the old versions and the information on who did what.

Git does that automatically, it did not infer this here due to additional chances, as it seems to me.

File: frontend/src/services/shortest-path/ShortestPathService.ts (hunk 1 above)
Comment (CatoLeanTruetschel) [reply frontend/src/services/shortest-path/ShortestPathService.ts:12]: Done!

File: frontend/src/stores/shortest-path/ShortestPathStateStore.ts (hunk 3 above)
Comment (CatoLeanTruetschel) [reply frontend/src/stores/shortest-path/ShortestPathStateStore.ts]: >why is this necessary?   

I added an additional default export, such that this is no longer necessary.  

>also add a short documentation what the store is responsible for  

Done!

File: frontend/src/stores/colors/EntityStyle.ts (hunk 2 above)
Comment (CatoLeanTruetschel) [reply frontend/src/stores/colors/EntityStyle.ts:4]: Done!

File: frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx (hunk 4 above)
Comment (CatoLeanTruetschel) [reply frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx]: Done!
